user-guide/Advanced_Modules/User Defined APIs/Tutorial/UD_APIs_Elements_in_alarm_tutorial.md

```diff
@@ -0,0 +1,120 @@
+   - In the window that opens, click 'Browse'. Now search for JSON, and click the first package (Newtonsoft.Json). In the right window, click 'Install'.
+
+   > [!NOTE]
+   > We could also use the built-in parameter conversion so we have them available as a Dictionary, how to do that is documented on the [defining an API page of the docs.](xref:UD_APIs_Define_New_API#user-input-data)  
+
+   ![Visual Studio project dependencies](~/user-guide/images/UDAPIS_elements_4.jpg)  
+
+      ![Visual Studio NuGet browse](~/user-guide/images/UDAPIS_elements_5.jpg)  
+  
+    ![Install Newtonsoft.Json](~/user-guide/images/UDAPIS_elements_6.jpg)  
+
+   - Now we'll parse our request body to an Input class that we'll make. We'll add error handling in case something else then we expect is being passed as request body.
+       - We're adding our own classes to map the JSON to so we have more control of the object that is being deserialized. This way, we're not depending on any API.
+   - We're adding a JsonSerializerSettings object with a ContractResolver, because the naming conventions for JSON is to use camelCasing, while our C# properties are in PascalCase.
+
+   [!code-csharp[](~/user-guide/Advanced_modules/User Defined APIs/Tutorial/Scripts/Elements_tutorial_script_C.cs?highlight=18-21,35-48,69-77)]
+
+   # [Step D](#tab/D)
+
+   In this step we're going to add the logic to our API that will get the Elements based on the alarmType passed in the request body. We'll put the logic in a separate method called `GetElements()`. We can wrap that method in a try-catch statement, and pass a user-friendly error in case something goes wrong.
+
+   In the method, we'll make sure the ElementFilter is correct to only show the elements with the alarm state that the user passed. We're mapping the Element objects returned by the automation engine to our own class, that is the class that we'll serialize to JSON and return to the user.
+
+   [!code-csharp[](~/user-guide/Advanced_modules/User Defined APIs/Tutorial/Scripts/Elements_tutorial_script_D.cs?highlight=61-73,77,82-118,124-135)]
+
+   # [Step E](#tab/E)
+
+   The last step is to publish the automation script to the DataMiner agent. If you need help with this, this is covered in the last item of step 2 in the [hello world tutorial](xref:UD_APIs_Hello_world_tutorial#step-2-create-the-automation-script).
+
+## Step 2: Create an API token
+
+It's good practice to not reuse API tokens. In the ideal scenario, you have a token for each consumer of your API. We're not going to reuse the token that we made in the hello world tutorial, we're going to make a new one. Don't remember how to? Consult [step 3 of the hello world tutorial](xref:UD_APIs_Hello_world_tutorial#step-3-create-an-api-token). Give it a name of your liking.
```

Comment:
I would not add too much text making reference to other tutorials. What you have done on step E above is perfect.

Reply:
These references could be added in a TIP block so they don't disrupt the flow of the text as much, but users who end up on a specific tutorial through the search can still find their way to the other tutorials when necessary.